src/Microsoft.ML.TorchSharp/Utils/DataUtils.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using TorchSharp;
+
+namespace Microsoft.ML.TorchSharp.Utils
+{
+    internal class DataUtils
+    {
+        public static torch.Tensor CollateTokens(IList<torch.Tensor> values, int padIndex, int? eosIndex = null,
+            bool leftPad = false, bool moveEosToBeginning = false, torch.Device device = null)
+        {
```

Comment:
can you assert `values.Count` is greater than 0?

Reply:
Added.